src/main/resources/native/ios/thread.m

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019, Gluon
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+ #include <stdio.h>
+
+void JVM_StartThread() {
```

Comment:
if you don't add this file, where does linking fail? 
It sounds strange we need all those symbols without usin them.

Reply:
Error without `thread.m`:

```
Undefined symbols for architecture arm64:
      "_JVM_CountStackFrames", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_CurrentThread", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_DumpThreads", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_GetAllThreads", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_HoldsLock", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_Interrupt", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_IsInterrupted", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_IsThreadAlive", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_ResumeThread", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_SetNativeThreadName", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_SetThreadPriority", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_Sleep", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_StartThread", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_StopThread", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_SuspendThread", referenced from:
          _methods in libjava.a(Thread.o)
      "_JVM_Yield", referenced from:
          _methods in libjava.a(Thread.o)
      "_Java_java_lang_Thread_countStackFrames", referenced from:
          _Thread_countStackFrames_62a502bde129f0997933886334feb845c363ba25 in llvm.o
      "_Java_java_lang_Thread_dumpThreads", referenced from:
          _Thread_dumpThreads_39b1e1fb2d2b0ace213e0ac1ec870e72f3e7d2ac in llvm.o
      "_Java_java_lang_Thread_getThreads", referenced from:
          _Thread_getThreads_891ea1885c7d2cf8e0d23e40e1463524f8c997ae in llvm.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    Linking failed
```

Reply:
So these references come from libjava.a, but why are those methods (e.g. _methods) included in linking? We don't use -Wl,load_all etc, so where would the references come from?
It seems unlikely to me there are indirect calls to Thread.o_methods.